wallet-new/test/DevelopmentSpec.hs

```diff
@@ -0,0 +1,78 @@
+{-# OPTIONS_GHC -fno-warn-orphans       #-}
+
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE NoImplicitPrelude          #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeSynonymInstances       #-}
+{-# LANGUAGE UndecidableInstances       #-}
+
+-- Spec for testing `development` endpoints
+module DevelopmentSpec where
+
+import           Universum
+
+import           Data.Default (def)
+import           Pos.Client.KeyStorage (addSecretKey, getSecretKeysPlain)
+
+import           Pos.Util.BackupPhrase (BackupPhrase (..), safeKeysFromPhrase)
+import           Pos.Util.CompileInfo (HasCompileInfo, withCompileInfo)
+import           Pos.Launcher (HasConfigurations)
+import           Pos.Util.QuickCheck.Property (assertProperty)
+
+import           Test.Pos.Wallet.Web.Mode (walletPropertySpec)
+import           Test.Hspec (Spec, describe)
+import           Test.Hspec.QuickCheck (modifyMaxSuccess)
+import           Test.Pos.Configuration (withDefConfigurations)
+
+import           Cardano.Wallet.API.Development.LegacyHandlers (deleteSecretKeys)
+import           Cardano.Wallet.Server.CLI (RunMode (..))
+import           Servant
+
+{-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
+
+spec :: Spec
+spec =
+    withCompileInfo def $
+    withDefConfigurations $
+    describe "development endpoint" $
+        describe "secret-keys" $ modifyMaxSuccess (const 10) deleteAllSecretKeysSpec
+
+deleteAllSecretKeysSpec :: (HasCompileInfo, HasConfigurations) => Spec
+deleteAllSecretKeysSpec = do
+    -- TODO: Use an arbitrary instance of `BackupPhrase` if available
+    let phrase = BackupPhrase [ "truly", "enact", "setup", "session"
+                              , "near", "film", "walk", "hard"
+                              , "ginger", "audit", "regular", "any"
+                              ]
+    walletPropertySpec "does remove all secret keys in debug mode mode" $ do
+        sKey <- either  (error "creating a secret key failed")
+                        (pure . fst)
+                        $ safeKeysFromPhrase mempty phrase
+        lift $ addSecretKey sKey
```

Comment:
@sectore Have you checked that running these specs will write and read secrets from a file which won't conflict with a production one? What I mean by that is that we should make sure that the `HasKeysContext` defined for the `WalletWebTestMode` monad is not using as keyfile `secrets.key`, as that is the same filename for "production" keys. Prudence is never enough..

Reply:
The endpoint is removing the `secrets.key` in `debug` mode only and not in `production` mode. That's why we do have this test to test this "feature".

Reply:
What I mean is that I'm not very comfortable in having test code which is deleting a file named exactly the same way it's named in production. It's easy to accidentally screw up a check for "production vs debug" and accidentally delete your own wallet's `secret.key`. It's probably paranoia (as you would never run a production node on the same folder you run the tests), but all I'm saying is that I would have been much happier if our tests would have run in a monad where the key file was named differently (dunno, `secrets-test.key`). Not a show-stopper, but more an out loud thought :)

Reply:
It might be a valid point ^. However, I'm not sure why `secret.key` could be deleted by running this `DevelopmentSpec`. At least I can't see any reference of `secret.key` in test code (either in `V0` nor in `V1` tests).